src/metabase/domain_entities/malli.cljc

```diff
@@ -19,47 +19,113 @@
 ;; hope. We can assume (deep) immutability so there's lots of room for caching of eg. key conversion.
 
 #?(:clj
-   (clojure.core/defn- argname [arg]
-     (cond
-       (symbol? arg)   arg
-       (and (map? arg)
-            (:as arg)) (:as arg)
-       :else           (gensym "arg"))))
+   (defmacro defn
+     "In Clojure this is simply [[mu/defn]].
+
+     That breaks CLJS advanced compilation since it returns a `let` and not a `defn`.
+     So in CLJS this is just [[clojure.core/defn]]."
+     [sym _ return-schema docs args & body]
+     (macros/case
+       :clj  `(mu/defn ~sym :- ~return-schema ~docs ~args ~@body)
+       :cljs `(clojure.core/defn ~sym ~docs ~(mapv first (partition 3 args)) ~@body))))
+
+(clojure.core/defn schema-for-path
+  "Given a schema and a *value path* (as opposed to a *schema path*), finds the schema for that
+  path. Throws if there are multiple such paths and those paths have different schemas."
+  [schema path]
+  (let [paths (-> schema mc/schema (mut/in->paths path))]
+    (cond
+      (empty? paths)      (throw (ex-info "Path does not match schema" {:schema schema :path path}))
+      (= (count paths) 1) (mut/get-in schema (first paths))
+      :else (let [child-schemas (map #(mut/get-in schema %) paths)]
+              (if (apply = child-schemas)
+                (first child-schemas)
+                (throw (ex-info "Value path has multiple schema paths, with different schemas"
+                                {:schema        schema
+                                 :paths         paths
+                                 :child-schemas child-schemas})))))))
 
 #?(:clj
```

Comment:
Dumb question but isn't a macro CLJ-only anyway?

Reply:
Yes, but if you don't guard them, they show up in the CLJS side as a regular function that can get into your REPL state, and if there's a function with that name it won't call the macro anymore when you reload.